packages/block-editor/src/components/block-list/block.js

```diff
@@ -372,34 +370,24 @@ function BlockListBlock( {
 				event.preventDefault();
 			}
 
-		// Avoid triggering multi-selection if we click toolbars/inspectors
-		// and all elements that are outside the Block Edit DOM tree.
-		} else if ( blockNodeRef.current.contains( event.target ) ) {
```

Comment:
Not sure why this condition is here... `onMouseDown` is already listens to the block-edit node.

Reply:
A long time ago, multiselection used to be triggered super easily while clicking on toolbars, link controls, modals... It might not be the case anymore but we should ensure that it's not being triggered inadvertently.